File: clients/kotlin/src/main/kotlin/model/LootBox.kt
package model

import util.StreamUtil

class LootBox {
    lateinit var position: model.Vec2Double
    lateinit var size: model.Vec2Double
    lateinit var item: model.Item
    constructor() {}
    constructor(position: model.Vec2Double, size: model.Vec2Double, item: model.Item) {
        this.position = position
        this.size = size
        this.item = item
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): LootBox {
            val result = LootBox()
            result.position = model.Vec2Double.readFrom(stream)
            result.size = model.Vec2Double.readFrom(stream)
            result.item = model.Item.readFrom(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        position.writeTo(stream)
        size.writeTo(stream)
        item.writeTo(stream)
    }
}

File: clients/scala/src/main/scala/model/LootBox.scala
package model

import util.StreamUtil

case class LootBox(position: model.Vec2Double, size: model.Vec2Double, item: model.Item) {
    def writeTo(stream: java.io.OutputStream) {
        position.writeTo(stream)
        size.writeTo(stream)
        item.writeTo(stream)
    }
}
object LootBox {
    def readFrom(stream: java.io.InputStream): LootBox = LootBox(
        model.Vec2Double.readFrom(stream)
        ,
        model.Vec2Double.readFrom(stream)
        ,
        model.Item.readFrom(stream)
        )
}
